app/models/user.rb

```diff
@@ -22,50 +20,12 @@ class User < ApplicationRecord
     length: {minimum: Settings.validate.password.length}
   validates :role, inclusion: {in: roles.keys}
 
-  before_save :downcase_email
-
-  has_secure_password
+  devise :database_authenticatable, :registerable, :rememberable, :validatable
```

Comment:
a add thêm giúp  1 vài cái nhỏ như
Thêm validate password phải không trùng với password cũ, passwod phải bao gồm (in hoa, in thường, số và phải từ 8-30 kí tự	
Ở màn reset password, phải cho nhập 1 trường current_password, nếu current_password đúng ms cho update	
Nếu login lỗi 5 lần liên tiếp thì khóa tài khoản trong 10 phút và gửi email về cho Admin thông tin login lỗi. Hết 10p nếu nhập đúng tài khoản sẽ đăng nhập bt
và tìm hiểu thêm về warden nhé

Reply:
> Thêm validate password phải không trùng với password cũ, passwod phải bao gồm (in hoa, in thường, số và phải từ 8-30 kí tự
> Ở màn reset password, phải cho nhập 1 trường current_password, nếu current_password đúng ms cho update

A add thêm mấy cái trên rồi nha e. Mấy cái phía dưới a sẽ tìm hiểu và làm ở PR gửi mail active luôn nha e :)))

> Nếu login lỗi 5 lần liên tiếp thì khóa tài khoản trong 10 phút và gửi email về cho Admin thông tin login lỗi. Hết 10p nếu nhập đúng tài khoản sẽ đăng nhập bt
> và tìm hiểu thêm về warden nhé